File: src/python/test/TestFlagged.py
import unittest
from Utils import Utils 
from Refi import Refi

class TestFlagged(unittest.TestCase):

    def setUp(self):
        self.refi = Refi()
        self.utils = Utils()


    def test_Flagged(self):

        expression = self.refi.toRegEx("`a` flagged { caseIns , `b` } `c`")
        self.utils.assertRegexIs("a(?i:b)c", expression)
        self.utils.assertRegexMatchesText(expression, "aBc")
        self.utils.assertRegexMatchesText(expression, "abc")
        self.utils.assertRegexDoesntMatchText(expression, "Axe") 
    
    def test_FlaggedComplex(self):

        expression = self.refi.toRegEx("`a` flagged { caseIns , (`b` digit) } `c`")
        self.utils.assertRegexIs("a(?i:(b\\d))c", expression)
        self.utils.assertRegexMatchesText(expression, "aB1c")
        self.utils.assertRegexMatchesText(expression, "ab1c")
        self.utils.assertRegexDoesntMatchText(expression, "Ax1e") 
    def runTest(self):
        pass  

if __name__ == '__main__':
    unittest.main()    
    

File: src/test/java/koorkah/TestFlagged.java
package koorkah;

import org.junit.jupiter.api.Test;

class TestFlagged extends BaseT
{

    @Test
    void testFlagged()
    {
        String expression = koorkah.toRegEx("`a` flagged { caseIns , `b` } `c`");
        assertRegexIs("a(?i:b)c", expression);
        assertRegexMatchesText(expression, "aBc");
        assertRegexMatchesText(expression, "abc");
        assertRegexDoesntMatchText(expression, "Axe"); 
    }
    
    @Test
    void testFlaggedComplex()
    {
        String expression = koorkah.toRegEx("`a` flagged { caseIns , (`b` digit) } `c`");
        assertRegexIs("a(?i:(b\\d))c", expression);
        assertRegexMatchesText(expression, "aB1c");
        assertRegexMatchesText(expression, "ab1c");
        assertRegexDoesntMatchText(expression, "Ax1e"); 
    }
}
